### PR TITLE
Add initial OpenWrt network configuration bits.

### PR DESCRIPTION
**Pull request by mrstegeman**

### CODE REVIEW

File: src/network-manager.js [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+          return false;
+        }
+      }
+
+      proc = child_process.spawnSync('uci', ['commit', 'network']);
+      if (proc.status !== 0) {
+        return false;
+      }
+
+      proc = child_process.spawnSync('/etc/init.d/network', ['reload']);
```
Comment (dhylands) [inline src/network-manager.js]: This should be `restart` not `reload`

File: src/network-manager.js [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+
+      let device = null;
+      let iface = null;
+      for (const line of proc.stdout.split('\n')) {
+        if (line.indexOf('=') < 0) {
+          continue;
+        }
+
+        const [key, value] = line.split('=', 2).map((s) => stripQuotes(s));
+
```
Comment (dhylands) [inline src/network-manager.js]: It feels like everything from spawnSync to here could be encapsulated into a common function?
Perhaps something that calls uci show and splits the results into lines?

File: src/network-manager.js (hunk 1 above)
Comment (mrstegeman) [reply src/network-manager.js]: The OpenWrt wiki disagrees: https://oldwiki.archive.openwrt.org/doc/uci/network

File: src/network-manager.js (hunk 2 above)
Comment (mrstegeman) [reply src/network-manager.js]: Agreed, this could be cleaned up a bit.

File: src/network-manager.js (hunk 1 above)
Comment (dhylands) [reply src/network-manager.js]: ok - I take it back. Running `/etc/init.d/network help` shows:
```
Syntax: /etc/init.d/network [command]

Available commands:
        start   Start the service
        stop    Stop the service
        restart Restart the service
        reload  Reload configuration files (or restart if service does not implement reload)
        enable  Enable service autostart
        disable Disable service autostart
```
and if I change the SSID then using reload does in fact change the SSID being advertised.

I guess I've just been used to using restart from the regular system V init scripts.

**Pull request merged**